### PR TITLE
Add Whitelist policy to CGroups Gateway

### PR DESCRIPTION
**Pull request by kursatkobya**
Whitelisting feauture implemented on cgroupparser
	Related unittests are added
	User dics has updated

### CODE REVIEW

File: libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp [hunk 1]
```diff
@@ -17,7 +22,28 @@ ReturnCode CGroupsParser::parseCGroupsGatewayConfiguration(const json_t *element
         return ReturnCode::FAILURE;
     }
 
-    result.first = settingKey;
-    result.second = settingValue;
+    log_debug() << "key : " << settingKey << " value : " << settingValue;
+
+    if ("memory.limit_in_bytes" == settingKey) {
+        if (m_settings.count(settingKey)) {
+            // if the new value is lexicographical greater than old one set is as the value
```
Comment (tobsan) [inline libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp]: This is not lexicographically comparison :)

**Review by tobsan — APPROVED** [review]
Looks good!

File: libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp (hunk 1 above)
Comment (kursatkobya) [reply libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp]: I just change it to integer comparision after deciding to apply to only one setting

File: libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp [hunk 2]
```diff
@@ -17,7 +22,28 @@ ReturnCode CGroupsParser::parseCGroupsGatewayConfiguration(const json_t *element
...
+            if (std::stoi(m_settings[settingKey]) < std::stoi(settingValue)) {
+                m_settings[settingKey] = settingValue;
+            }
+        } else {
+            m_settings[settingKey] = settingValue;
+        }
+    } else {
+        log_warning() << "It is possible to apply this rule through control groups of working "
```
Comment (breakreturn) [inline libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp]: I think this should be more specific and clear, e.g. "cgroups gateway config \"" <<  config << "\" is not supported by the gateway"

**Review by breakreturn — CHANGES_REQUESTED** [review]

File: libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp (hunk 2 above)
Comment (kursatkobya) [reply libsoftwarecontainer/src/gateway/cgroups/cgroupsparser.cpp]: But kernel can support this setting so maybe it is not told that its not supported. But we can keep saying this setting is not guaranteed to work by cgroup gateway. But i agree we should mentioned about the key. Thus i propose to keep sentence but changing "this rule" part with the actual setting key.

**Review by DunderRoffe — APPROVED** [review]

**Review by breakreturn — APPROVED** [review]

**Pull request merged**